### PR TITLE
TG-791 | payment receipt document

### PR DESCRIPTION
**Pull request by Yousef-Hammar**

### CODE REVIEW

File: broker/payment-receipt.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func paymentReceiptBuilder(transactionUID string, authToken lib.AuthToken) (string, string, error) {
+	transaction := trx.GetTransactionByUid(transactionUID, "")
+	if transaction == nil {
+		return "", "", errors.New("transaction not found")
+	}
+
+	if !transaction.IsPay {
+		return "", "", errors.New("transaction is not pay")
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: pa**id**

File: broker/payment-receipt.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	// TODO: what if transaction refers to a renewPolicy
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: do we foresee that the api will be used from the renew tab? If so we might have to add a query param to indicate.

File: broker/payment-receipt.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+			return "", "", fmt.Errorf("node %s cannot access policy %s", authToken.UserID, policy.Uid)
+		}
+
+		node := network.GetNetworkNodeByUid(policy.ProducerUid)
+		if node == nil {
+			return "", "", errors.New("node not found")
+		}
+
+		if policy.ProducerUid != node.Uid && !network.IsChildOf(node.Uid, policy.ProducerUid) {
+			return "", "", fmt.Errorf("node %s cannot access policy %s", authToken.UserID, policy.Uid)
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: check for plcUtils.CanBeAccessedBy

File: broker/payment-receipt.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	expirationDate := lib.AddMonths(transaction.EffectiveDate, 12)
+	nextPayment := lib.AddMonths(transaction.EffectiveDate, 12)
+	if policy.PaymentSplit == string(models.PaySplitMonthly) {
+		expirationDate = lib.AddMonths(transaction.EffectiveDate, 1).AddDate(0, 0, -1)
+		nextPayment = lib.AddMonths(transaction.EffectiveDate, 1)
+	}
+
+	payDate := "======="
+	if !transaction.PayDate.IsZero() {
+		payDate = transaction.PayDate.Format("02/01/2006")
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: save the format as a constant

File: document/payment-receipt.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	pdf.MultiCell(0, 4, text, "", fpdf.AlignLeft, false)
+
+	pdf.Ln(5)
+
+	text = "La ringraziamo e Le porgiamo i nostri più cordiali saluti."
+	pdf.MultiCell(0, 4, text, "", fpdf.AlignLeft, false)
+
+	pdf.Ln(17)
+
+	text = "RICEVUTO DI PAGAMENTO DEL PREMIO"
```
Comment (diogo-wopta) [inline document/payment-receipt.go]: Haven't seen the requirements and could be an error on my part, but shouldn't it be RICEVUT**A** ?

File: broker/payment-receipt.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+		Email:      policy.Contractor.Mail,
+		Phone:      policy.Contractor.Phone,
+	}
+
+	expirationDate := lib.AddMonths(transaction.EffectiveDate, 12)
+	nextPayment := lib.AddMonths(transaction.EffectiveDate, 12)
+	if policy.PaymentSplit == string(models.PaySplitMonthly) {
+		expirationDate = lib.AddMonths(transaction.EffectiveDate, 1).AddDate(0, 0, -1)
+		nextPayment = lib.AddMonths(transaction.EffectiveDate, 1)
+	}
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: for the nextPayment we need to check the policy end date.

File: broker/payment-receipt.go [hunk 7]
```diff
@@ -127,24 +157,35 @@ func receiptInfoBuilder(policy models.Policy, transaction models.Transaction) do
...
+	if !tmpNextPayment.After(policy.EndDate) {
+		nextPayment = tmpNextPayment.Format(dateFormat)
 	}
 
 	payDate := "======="
 	if !transaction.PayDate.IsZero() {
-		payDate = transaction.PayDate.Format("02/01/2006")
+		payDate = transaction.PayDate.Format(dateFormat)
 	}
```
Comment (diogo-wopta) [inline broker/payment-receipt.go]: Check if for the policy end date we have to subtract a day or not.

In the case of monthly policies we have a date for the expiry and the following day is the next payment day. But for yearly the dates are the same.

Comment (Yousef-Hammar) [reply broker/payment-receipt.go]: I talked to Linda, and she told me that expiration date is effective date +1 month/year based on payment split

**Review by diogo-wopta — APPROVED** [review]

**Pull request merged**